packages/composable-controller/tsconfig.json

```diff
@@ -1,7 +1,8 @@
 {
   "extends": "../../tsconfig.packages.json",
   "compilerOptions": {
-    "baseUrl": "./"
+    "baseUrl": "./",
+    "lib": ["ES2019", "DOM"]
```

Comment:
Updated because AddressBookController uses an ES2019 feature (`flatMap`).

Reply:
Using `flatMap` will break on the extension as well, because the minimum Chromium version there is still v66 (which does not support `flatMap`).

We could use lodash's `flatmap` instead. That's what the extension uses.

Reply:
Right, but shouldn't this get compiled down? `lib` just informs TypeScript about this method, but the `target` is the thing that controls how that gets compiled.

Reply:
That works for syntax, not for global APIs. TypeScript doesn't supply its own polyfills, we'd have to add our own.

Reply:
Sure, that makes sense.

Hmm. I feel like this is the kind of thing we ought to catch early, because it can be easily missed. In other words, I want to see something in the lint, build, or test step fail if we attempt to use an feature from an ES version that exceeds our max supported version. This is out of scope obviously, but what do you think about running the compiled JavaScript we get from TypeScript through Babel? Or — and this is something I've thought about recently — what if we require `yarn build` to be run before running tests, so that the tests are run against the compiled JavaScript rather than being compiled on the fly? I guess they would still be getting run in a Node environment, though, so it would depend on the runtime. Maybe the only true way to test this would to run the tests in a browser. Hmm.

Reply:
Well, to answer my own question, I guess the `lib` setting works for now, maybe we ought to just document it better.

Reply:
I'm hoping this requirement will go away soon when we update the minimum browser we support to one that includes ES2020 support., and put systems in place to make it easier to update our minimum version going forward. That would let us keep our libraries reasonably well ahead in terms of which targets we use.